include/dslash_helper.cuh

```diff
@@ -349,26 +351,26 @@ namespace quda
       retcount_inter(dslash::get_shmem_retcount_inter())
 #endif
     {
-      if (in.data() == out.data()) errorQuda("Aliasing pointers");
+      for (auto i = 0u; i < in.size(); i++)
```

Comment:
Not that I want to open a quadratic scaling can of worms, but wouldn't it be a problem if anything in `in` aliased with anything in `out`? (...or anything in `out` with anything else in `out`)